llvm-to-smt/cocci/remove_same_reg_comparison.cocci

```diff
@@ -0,0 +1,20 @@
+@remove_same_reg_branch@
+identifier fn = is_scalar_branch_taken;
+@@
+
+fn(...) {
+<...
+-	if (reg1 == reg2) { ... }
+...>
+}
+
+@remove_same_reg_skip_refine@
+identifier fn = reg_set_min_max;
+@@
+
+fn(...) {
+<...
+-	if (false_reg1 == false_reg2)
+-		return 0;
```

Comment:
I wonder if we could make this more generic so that variable names don't matter.